Makefile

```diff
@@ -1,33 +1,41 @@
+# variables that should not be overridden by the user
 VERSION = edge
-TAG = $(VERSION)
-PREFIX ?= nginx-kubernetes-gateway
-
-GIT_COMMIT = $(shell git rev-parse HEAD)
+GIT_COMMIT = $(shell git rev-parse HEAD || echo "unknown")
 DATE = $(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
 
-TARGET ?= local
-
-KIND_KUBE_CONFIG_FOLDER = $${HOME}/.kube/kind
-OUT_DIR=$(shell pwd)/build/.out
+# variables that can be overridden by the user
+PREFIX ?= nginx-kubernetes-gateway## The name of the image. For example, nginx-kubernetes-gateway
+TAG ?= $(VERSION:v%=%)## The tag of the image. For example, 0.3.0
+TARGET ?= local## The target of the build. Possible values: local and container
+KIND_KUBE_CONFIG_FOLDER = $${HOME}/.kube/kind## The folder where the kind kubeconfig is stored
```

Comment:
Do we really need `KIND_KUBE_CONFIG_FOLDER`?

Reply:
why not? what is the alternative?

Reply:
I'm just wondering if it needs to be configurable. Personally I always use the default `~/.kube/config`.

Reply:
so do we keep it or not? 🙂

Reply:
Probably don't need it to be configurable right now.